### ISSUE TITLE
Make "working_dir" of volume relative paths configurable via command line flag or ENV var

### ISSUE THREAD

**Issue by pchico83** [docker/compose#1700]
This configuration setting would be used to resolve relative paths in the "volumes" property. Our use case is:

- Container is running a build after `git clone` a repository.
- This container is mounting `/var/run/docker.sock` to make use of docker cache between builds.
- The build container does `docker-compose up`, but relative paths don't work as containers are created by the host docker daemon (it would have work in a docker-in-docker environment).

**Comment by pchico83** [comment]
I can fixed in my fork like this:

```
def resolve_volume_paths(volumes, working_dir=None):
    if working_dir is None:
        raise Exception("No working_dir passed to resolve_volume_paths()")
    current_dir = os.path.abspath(working_dir)
    if working_dir.startswith(current_dir) and os.environ.get('COMPOSE_BASE_DIR'):
        working_dir = os.path.join(os.environ.get('COMPOSE_BASE_DIR'), working_dir[len(current_dir) + 1:])

    return [resolve_volume_path(v, working_dir) for v in volumes]
```

but there is probably a more elegant solution.

**Comment by aanand** [comment]
Can you describe your use case a bit more fully, perhaps with an example `docker-compose.yml` and usage?

**Comment by pchico83** [comment]
Sure, I have a `docker-compose.yml` file like this:

```
sut:
  build: .
  command: /app/run_tests.sh
example:
  image: tutum/helloworld
  volumes:
    - "./certificate:/etc/certificate"
```
where the service `example` is mounting "volumes" from a relative path. 

Then, we trigger a container for every Github commit that mounts `/var/run/docker.sock` and does:

```
    cd /app
    git clone commit
    docker-compose -p app up sut
    RET=$(docker wait app_sut_1)
    if [ "$RET" != "0" ]; then
         echo "   Tests FAILED: $RET"
    else
         echo "   Tests PASSED"
    fi
 ```

Now, `docker-compose` expands the relative path in the service `example` from `./certificate` to `/app/certificate`, but the container is created by the host docker daemon (we are mounting `/var/run/docker.sock` to take advantage of docker cache between different git commits) and everything breaks because that path does not exist.

I would like to make `/app` a configurable parameter (or by ENV var) to tell `docker-compose` to expand `.` with `/var/lib/docker/aufs/mnt/4a2...63c/app`, which is the host path for `.`. It is a kind of hack, but I have been testing it and it would work.
Another solution would be to support "host volumes" in `docker-compose` when the docker daemon is a remote server, but I don't know what are the plans for this in `docker-compose`.

**Comment by aanand** [comment]
You can expand environment variables in volume paths (it's an undocumented feature because it's not implemented very well - see #1377 for the master plan):

```yaml
volumes:
  - ${CERT_PATH}:/etc/certificate
```

That should cover your use case, if I've understood it correctly.

I'm strongly against making the meaning of relative paths (without env vars in them) mutable - it makes it much harder to read a YAML file and know what it's going to do.

**Comment by pchico83** [comment]
Ok, that makes sense. Thanks!
